implementations/rust/ockam/ockam_api/src/cloud/project/controller_client.rs

```diff
@@ -72,6 +74,126 @@ impl ControllerClient {
             .miette_success("get orchestrator version")
     }
 
+    pub async fn add_space_admin(
+        &self,
+        ctx: &Context,
+        space_id: &str,
+        email: &EmailAddress,
+    ) -> miette::Result<AdminInfo> {
+        trace!(target: TARGET, "adding new space administrator");
+        let admins: Vec<AdminInfo> = self
+            .get_secure_client()
+            .ask(
+                ctx,
+                "spaces",
+                Request::put(format!("/v0/{space_id}/admins/{email}")),
+            )
+            .await
+            .into_diagnostic()?
+            .miette_success("add space admins")?;
+        admins
+            .into_iter()
+            .find(|a| a.email == email.to_string())
+            .ok_or(miette!("The email {email} was not added to the space"))
+    }
+
+    pub async fn list_space_admins(
+        &self,
+        ctx: &Context,
+        space_id: &str,
+    ) -> miette::Result<Vec<AdminInfo>> {
+        trace!(target: TARGET, "listing space administrators");
+        self.get_secure_client()
+            .ask(
+                ctx,
+                "spaces",
+                Request::get(format!("/v0/{space_id}/admins")),
+            )
+            .await
+            .into_diagnostic()?
+            .miette_success("get space admins")
+    }
+
+    pub async fn delete_space_admin(
+        &self,
+        ctx: &Context,
+        space_id: &str,
+        email: &EmailAddress,
+    ) -> miette::Result<()> {
+        trace!(target: TARGET, "deleting space administrator");
+        let _admins: Vec<AdminInfo> = self
+            .get_secure_client()
+            .ask(
+                ctx,
+                "spaces",
+                Request::delete(format!("/v0/{space_id}/admins/{email}")),
+            )
+            .await
+            .into_diagnostic()?
+            .miette_success("delete space admins")?;
```

Comment:
We would return an error 500 if it fails on our end. Otherwise we have the status provided on the elixir side, but looks like we are not using that when parsing the `MietteError` into another of our error types.

Reply:
Maybe this should be part of another PR but right now the output is:
```
❯ ockam project-admin delete me2@email.com


    Error:
    ────────────────────────────────────────────────────────────────────────────────

    The project admin me2@email.com was not found
    Please report this issue, with a copy of your logs, to https://github.com/build-trust/ockam/issues

    Error code: OCK500
    version: 0.129.0
    compiled from: 49252157cb3537a82a0af0d2275bfdff1092c7a1
```
And we shouldn't tell the user to open an issue for this.

Reply:
oooh right. I think this is easier to fix as this error is created [in the comand](https://github.com/build-trust/ockam/blob/49252157cb3537a82a0af0d2275bfdff1092c7a1/implementations/rust/ockam/ockam_command/src/terminal/tui.rs#L175). Let me have a look.

Reply:
There is a tradeoff here due to the error issue... Because of the way we bubble up errors (plain miette errors) we can't say what kind of error we had, so 99% of the times we are converting the error into a code 500. If we want to return a plain message without the error template, then we would have to return an Ok(), which would be wrong from the command perspective (e.g. when using the command in a script we check the exit code).

I wouldn't change anything for now until we refactor the error handling.

Reply:
Makes sense.